src/ProductService.java

```diff
@@ -38,4 +38,21 @@ public Product getProductById(int id) {
         }
```

Comment:
Here are my suggestions for best coding practices and semantic convention for the given functions:

1. `getProductById` function:

- The function should be declared `public` to allow access from other classes.
- The function should have a clear purpose and name that reflects its functionality, i.e. `getProductById`.
- The function should have a single responsibility and should only perform the task of retrieving a product based on its ID.
- The function should return a `Product` object or `null` if the product with the specified ID was not found.
- The function should include appropriate error handling to prevent exceptions from being thrown.

2. `convertToHex` function:

- The function should be declared `public static` to allow access from other classes.
- The function should have a clear purpose and name that reflects its functionality, i.e. `convertToHex`.
- The function should have a single responsibility and should only perform the task of converting a byte array to a hex string.
- The function should use a loop to iterate over each byte in the input array and convert it to a hex string using appropriate logic.
- The function should return the hex string.
- The function should include appropriate error handling to prevent exceptions from being thrown.